in_toto/runlib.py

```diff
@@ -487,34 +522,70 @@ def in_toto_record_start(step_name, key, material_list):
 
   link_metadata = Metablock(signed=link)
 
-  log.info("Signing link metadata with key '{:.8}...'...".format(key["keyid"]))
-  link_metadata.sign(key)
+  if signing_key:
+    log.info("Signing link metadata using passed key...")
+    signature = link_metadata.sign(signing_key)
+
+  elif gpg_keyid:
```

Comment:
I think we can take the `else` away by having `elif gpg_keyid or gpg_use_default:`

Reply:
Or even `else: # gpg_keyid or gpg_use_default`, c.f. https://github.com/in-toto/in-toto/pull/167#discussion_r160774560 :)

Reply:
Oh, the reason why I did it that way was to have different log statements. "*using passed GPG keyid"* vs. "*using default GPG key"*.
I could put the function call which is the same for both in an `else`, and have an additional `if/else` inside for the log statement, but I don't think that that would increase readability.

Reply:
Another option is to relocate the log statement to `sign_gpg()`.  It might be easier to handle the "GPG keyid" vs "default GPG key" case there.

Note: The guidelines recommend never to use short circuit evaluation to produce output.
https://github.com/secure-systems-lab/code-style-guidelines#never-use-short-circuit-evaluation-to-produce-output

Reply:
Hmm, I think the line is blurry on what is "short circuit evaluation" then. This is just an `or` clause. Am I missing something?

Reply:
re "custom message for default keyid". I don't know how useful is it really. We can leave it though...

Reply:
Alright, I added a pointer https://github.com/in-toto/in-toto/issues/117#issuecomment-356756728. Thanks for the review!